server/src/main/java/com/google/android/attestation/ParsedAttestationRecord.java

```diff
@@ -96,9 +96,27 @@ private ParsedAttestationRecord(
     this.teeEnforced = teeEnforced;
   }
 
-  public static ParsedAttestationRecord createParsedAttestationRecord(X509Certificate cert)
+  public static ParsedAttestationRecord createParsedAttestationRecord(X509Certificate[] certs)
       throws IOException {
-    ASN1Sequence extensionData = extractAttestationSequence(cert);
+
+    // The last seen certificate with an attestation is considered benign (i.e.,
```

Comment:
I know the attack being prevented but couldn't tell from this comment. Could you change it to be more like "Parse the attestation record that is closest to the root. This prevents an adversary from attesting an attestation record of their choice with an otherwise trusted chain using the following attack: 1) having the TEE attest a key under the adversary's control, 2) using that key to sign a new leaf certificate with an attestation extension that has their chosen attestation record, then 3) appending that certificate to the original certificate chain."